### ISSUE TITLE
TypeError: Cannot read property 'event' of undefined

### ISSUE THREAD

**Issue by Denton-L** [ethereum/browser-solidity#115]
Whenever a contract is called by another contract, if the callee raises an event which is not defined by the caller, an error is emitted, `TypeError: Cannot read property 'event' of undefined`. This seems to be an issue with browser-solidity because this issue is not present on identical code not running on browser-solidity.

Example code:

```
contract A {
    event eventA();
    
    function makeEvent() {
        eventA();
    }
}


contract B {
    function callA() {
        A a = new A();
        a.makeEvent(); // causes an error in browser-solidity
    }
}
```

**Comment by redsquirrel** [comment]
Without looking more deeply into it, the one thing that stands out is the `constant` keyword on `makeEvent`. Was that intentional? Doesn't `constant` imply that there are no side-effects of calling that function?

**Comment by Denton-L** [comment]
Good catch, I edited the code to remove the `constant`. But yes, this still breaks it.

**Comment by chriseth** [comment]
`constant` is not yet enforced but I guess events should be allowed because those do not change the state.

**Comment by zmitton** [comment]
same error message, but possibly for a different reason. As far as I can tell events are no longer working at all. Similar code to this seemed to work last week. Here is some code I boiled down:

    contract Thing{
        event LogSomething(address sender);
    
        function logSomething(){
            LogSomething(msg.sender);
        }
    }
This gives me the same error code stated above "TypeError: Cannot read property 'event' of undefined"

**Comment by ah903** [comment]
No idea why but declaring the events in client contracts seems to work. Presumably would also work if events were defined in some base class but haven't tried this. Some examples that seem to work on 0.3.6 solidity compiler

contract A {
    event LogStuff(bytes32 msg);
    function Log(bytes32 msg){
        LogStuff(msg);
    }
}

contract B {
    //Required to log an event through a.Log()
    // otherwise throws exception
    event LogStuff(bytes32 msg);
    
    function Run(){
        A a = new A();
        a.Log("Test");
    }
}

contract C{
    
    //Required to log an event through a.Log()
    event LogStuff(bytes32 msg);
 
    A a;
    
    function C(A ainst){
        a=ainst;
    }
    
    function Run(){
        a.Log("Test");
    }
}

contract D is B{
    function Base(){
        Run();
    }
}

contract E{
    
    B b;
    function E(B binst){
        b=binst;
    }
    
    function Run(){
        b.Run();
    }
}

**Comment by D-Nice** [comment]
Also getting 
> TypeError: Cannot read property 'event' of undefined

for any events, even the simplest. Here's the test contract I'm using:

    contract EventsTest {
        event LogTest(address _this);
    
        function EventsTest() {
           LogTest(this);    
        }
    }

**Comment by ah903** [comment]
I think the issue is confined to the online browser compiler. Once deployed all seems fine although I've only tested with testrpc so far. I also found issue to be intermittent. Sometimes refreshing the browser and trying again generated the event sometimes not.

**Comment by axic** [comment]
Fixed in #190.
